File: api/csharp/ToolGood.TextFilter.Api/Impl/KeywordTypeProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using ToolGood.TextFilter.Api.Datas;
using ToolGood.TextFilter.Api.Datas.KeywordTypes;
using ToolGood.TextFilter.Api.Interfaces;

namespace ToolGood.TextFilter.Api.Impl
{
    public class KeywordTypeProvider : ProviderBase, IKeywordTypeProvider
    {
        public KeywordTypeProvider() : base(TextFilterConfig.Instance) { }
        public KeywordTypeProvider(TextFilterConfig textFilterConfig) : base(textFilterConfig) { }


        private const string GetListUrl = "/api/get-keywordtype-list";
        private const string SetKeywordTypeUrl = "/api/set-keywordtype";

        public KeywordtypeListResult GetList()
        {
            return GetContent<KeywordtypeListResult>(GetListUrl);
        }
        public Task<KeywordtypeListResult> GetListAsync()
        {
            return GetContentAsync<KeywordtypeListResult>(GetListUrl);
        }


        public CommonResult SetKeywordType(int typeId, int level_1_UseType, int level_2_UseType, int level_3_UseType
            , bool useTime = false, string startTime = "", string endTime = "")
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["typeId"] = typeId;
            dict["level_1_UseType"] = level_1_UseType;
            dict["level_2_UseType"] = level_2_UseType;
            dict["level_3_UseType"] = level_3_UseType;
            dict["useTime"] = useTime;
            dict["startTime"] = startTime;
            dict["endTime"] = endTime;
            var json = JsonConvert.SerializeObject(dict);
            return PostContent<CommonResult>(SetKeywordTypeUrl, json);
        }
        public Task<CommonResult> SetKeywordTypeAsync(int typeId, int level_1_UseType, int level_2_UseType, int level_3_UseType
            , bool useTime = false, string startTime = "", string endTime = "")
        {
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["typeId"] = typeId;
            dict["level_1_UseType"] = level_1_UseType;
            dict["level_2_UseType"] = level_2_UseType;
            dict["level_3_UseType"] = level_3_UseType;
            dict["useTime"] = useTime;
            dict["startTime"] = startTime;
            dict["endTime"] = endTime;
            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<CommonResult>(SetKeywordTypeUrl, json);
        }



    }
}

File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Impl/KeywordTypeProvider.java
package toolgood.textfilter.api.Impl;

import java.util.Hashtable;
import java.util.Map;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import toolgood.textfilter.api.TextFilterConfig;
import toolgood.textfilter.api.Datas.CommonResult;
import toolgood.textfilter.api.Datas.KeywordTypes.KeywordtypeListResult;
import toolgood.textfilter.api.Interfaces.IKeywordTypeProvider;

public class KeywordTypeProvider extends ProviderBase implements IKeywordTypeProvider {

    public KeywordTypeProvider(TextFilterConfig textFilterConfig) {
        super(textFilterConfig);
    }

    private final String GetListUrl = "/api/get-keywordtype-list";
    private final String SetKeywordTypeUrl = "/api/set-keywordtype";

    @Override
    public KeywordtypeListResult GetList() {
        String content = doGet(GetListUrl);

        Gson gson = new GsonBuilder().create();
        KeywordtypeListResult result = gson.fromJson(content, KeywordtypeListResult.class);
        return result;
    }

    @Override
    public CommonResult SetKeywordType(int typeId, int level_1_UseType, int level_2_UseType, int level_3_UseType,
            Boolean useTime, String startTime, String endTime) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("typeId", typeId);
        dictionary.put("level_1_UseType", level_1_UseType);
        dictionary.put("level_2_UseType", level_2_UseType);
        dictionary.put("level_3_UseType", level_3_UseType);
        dictionary.put("useTime", useTime);
        dictionary.put("startTime", startTime);
        dictionary.put("endTime", endTime);
        String content = doPost(SetKeywordTypeUrl, dictionary);

        Gson gson = new GsonBuilder().create();
        CommonResult result = gson.fromJson(content, CommonResult.class);
        return result;
    }

}
